sdk/src/types/block/core/basic.rs

```diff
@@ -0,0 +1,239 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use packable::{
+    error::{UnpackError, UnpackErrorExt},
+    packer::Packer,
+    unpacker::Unpacker,
+    Packable,
+};
+
+use crate::types::block::{
+    core::{verify_parents, Block},
+    parent::{ShallowLikeParents, StrongParents, WeakParents},
+    payload::{OptionalPayload, Payload},
+    protocol::ProtocolParameters,
+    Error,
+};
+
+/// A builder for a [`BasicBlock`].
+pub struct BasicBlockBuilder {
+    strong_parents: StrongParents,
+    weak_parents: WeakParents,
+    shallow_like_parents: ShallowLikeParents,
+    payload: OptionalPayload,
+    burned_mana: u64,
+}
+
+impl BasicBlockBuilder {
+    /// Creates a new [`BasicBlockBuilder`].
+    #[inline(always)]
+    pub fn new(strong_parents: StrongParents, burned_mana: u64) -> Self {
+        Self {
+            strong_parents,
+            weak_parents: WeakParents::default(),
+            shallow_like_parents: ShallowLikeParents::default(),
+            payload: OptionalPayload::default(),
+            burned_mana,
+        }
+    }
+
+    /// Adds strong parents to a [`BasicBlockBuilder`].
+    #[inline(always)]
+    pub fn with_strong_parents(mut self, strong_parents: impl Into<StrongParents>) -> Self {
```

Comment:
strong and burned mana setters arent needed. (Validation block has the new() params not as setters)

Reply:
Yeah, for instance if you create a builder from an existing output

Reply:
Then lets add the setters to validation block as well

Reply:
Done